apps/04TowersOfHanoi.js

```diff
@@ -19,24 +19,31 @@ function printStacks() {
   console.log("c: " + stacks.c);
 }
 
-function movePiece() {
+function movePiece(startStack, endStack) {
   // Your code here
-
+  var movedItem = stacks[startStack].pop();
+  stacks[endStack].push(movedItem);
 }
 
-function isLegal() {
+function isLegal(startStack, endStack) {
   // Your code here
-
+  return !(stacks[startStack][stacks[startStack].length - 1] > stacks[endStack][stacks[endStack].length - 1])
```

Comment:
Amazing concatenation! For readability, try breaking up at least some of your code in variables.

Ex: 
var startIndex = stacks[startStack].length - 1;
...
...
return !(stacks[startStack][startIndex] < stacks[startStack][endIndex]);